test/with_dummyserver/test_socketlevel.py

```diff
@@ -842,3 +842,67 @@ def test_empty_head_response_does_not_hang(self):
 
         # stream will use the read method here.
         self.assertEqual([], list(r.stream()))
+
+
+class TestChunkedTransfer(SocketDummyServerTestCase):
+    def _start_chunked_handler(self):
+        self.buffer = b''
+
+        def socket_handler(listener):
+            sock = listener.accept()[0]
+
+            while not self.buffer.endswith(b'\r\n0\r\n\r\n'):
+                self.buffer += sock.recv(65536)
+
+            sock.send(
+               b'HTTP/1.1 200 OK\r\n'
+               b'Content-type: text/plain\r\n'
+               b'Content-Length: 0\r\n'
+               b'\r\n')
+            sock.close()
+
+        self._start_server(socket_handler)
+
+    def test_chunks(self):
+        self._start_chunked_handler()
+        chunks = ['foo', 'bar', '', 'bazzzzzz']
+        pool = HTTPConnectionPool(self.host, self.port, retries=False)
+        r = pool.urlopen('GET', '/', chunks, headers=dict(DNT='1'), chunked=True)
+
+        self.assertTrue(b'Transfer-Encoding' in self.buffer)
+        body = self.buffer.split(b'\r\n\r\n', 1)[1]
+        lines = body.split(b'\r\n')
+        # Empty chunks should have been skipped, as this could not be distinguished
+        # from terminating the transmission
+        for i, chunk in enumerate([c for c in chunks if c]):
+            self.assertEqual(lines[i * 2], str(len(chunk)).encode('utf-8'))
+            self.assertEqual(lines[i * 2 + 1], chunk.encode('utf-8'))
+
+    def test_string_body(self):
+        self._start_chunked_handler()
+        pool = HTTPConnectionPool(self.host, self.port, retries=False)
+        body_str = 'thisshouldbeonechunk'
+        r = pool.urlopen('GET', '/', body_str, chunked=True)
+        header, body = self.buffer.split(b'\r\n\r\n', 1)
+
+        self.assertTrue(b'Transfer-Encoding: chunked' in header.split(b'\r\n'))
+        self.assertTrue(b'\r\n' + body_str.encode('utf-8') + b'\r\n' in body)
+        self.assertTrue(body.endswith(b'\r\n0\r\n\r\n'))
```

Comment:
I basically tried to keep the body_str variable. If I check for a fixed output, I also have to reimplement the encoding e.g. of the len of the chunk etc.

In general I'd be a big fan of parametrizing tests, so this may be run e.g. also with a randomly generated 1k string, or 100k, just to make sure it doesn't break. Though, unittest sucks at parametrizing, that's why I skipped it.